ddl/ddl_api.go

```diff
@@ -2192,6 +2192,10 @@ func modifiableCharsetAndCollation(toCharset, toCollate, origCharset, origCollat
 // It returns true if the two types has the same Charset and Collation, the same sign, both are
 // integer types or string types, and new Flen and Decimal must be greater than or equal to origin.
 func modifiable(origin *types.FieldType, to *types.FieldType) error {
+	// Because EncodeDecimal encodes the precision information, if change the decimal precision, need rewrite the record.
```

Comment:
The root cause is modifying decimal precision needs to rewrite binary representation of that decimal?

Reply:
done.